R/sitevisitpeak-data.R

```diff
@@ -14,88 +14,100 @@
+      timeFormatting[[1]] <- sapply(timeFormatting[[1]], function(s) sub(".000","",s))
+      timeFormatting[[2]] <- paste(" (UTC",timeFormatting[[2]], ")")
+      timeFormatting <-  paste(timeFormatting[[1]],timeFormatting[[2]])
+      
+      if(!is.null(listElements$estimatedTime) && !is.na(listElements$estimatedTime)) {
+        estDateTime <- (strsplit(listElements$estimatedTime, split="[T]"))
+        estDate <- strftime(estDateTime[[1]][1], "%m/%d/%Y")
+      } else {
+        estDate <- ""
+      }
+      
+      
+      if(!is.null(listElements$associatedIvTime) && !is.na(listElements$associatedIvTime)) {
+        ivDateTime <- (strsplit(listElements$associatedIvTime, split="[T]"))
+        ivDate <- strftime(ivDateTime[[1]][1], "%m/%d/%Y")
+        
+        ivTimeFormatting <- sapply(ivDateTime[[1]][2], function(s) strsplit(s,split="[-]")[[1]])
+        ivTimeFormatting[[1]] <- sapply(ivTimeFormatting[[1]], function(s) sub(".000","",s))
+        ivTimeFormatting[[2]] <- paste(" (UTC",ivTimeFormatting[[2]], ")")
+        ivTimeFormatting <-  paste(ivTimeFormatting[[1]],ivTimeFormatting[[2]])
+      } else {
+        ivDate <- ""
+        ivTimeFormatting <- ""
+      }
+      
+      quals <- getQualifiers(listElements$associatedIvTime, listElements$associatedIvQualifiers)
+      
+      diff <- getIvDifference(listElements$value, listElements$associatedIvValue)
+      
+      toAdd = c(date,
+                timeFormatting,
+                nullMask(listElements$party), 
+                nullMask(listElements$sublocation), 
+                nullMask(listElements$monitoringMethod), 
+                nullMask(listElements$value),
+                nullMask(listElements$uncertainty), 
+                estDate, 
+                nullMask(listElements$comments),
+                nullMask(listElements$associatedIvValue),
+                quals,
+                ivDate,
+                ivTimeFormatting,
+                diff)
+      
+      toRet <- rbind(toRet, data.frame(t(toAdd),stringsAsFactors = FALSE))
     }
+    colnames(toRet) <- columnNames
+    rownames(toRet) <- NULL
     
-    quals <- getQualifiers(listElements$associatedIvTime, listElements$associatedIvQualifiers)
-    
-    diff <- getIvDifference(listElements$value, listElements$associatedIvValue)
-    
-    toAdd = c(date,
-              timeFormatting,
-              nullMask(listElements$party), 
-              nullMask(listElements$sublocation), 
-              nullMask(listElements$monitoringMethod), 
-              nullMask(listElements$value),
-              nullMask(listElements$uncertainty), 
-              estDate, 
-              nullMask(listElements$comments),
-              nullMask(listElements$associatedIvValue),
-              quals,
-              ivDate,
-              ivTimeFormatting,
-              diff)
-    
-    toRet <- rbind(toRet, data.frame(t(toAdd),stringsAsFactors = FALSE))
+    return(toRet)
+  }  
+  if(length(data) == 0) {
```

Comment:
on else it complains and won't build. but else if works.. is that acceptable or should I see why the else alone is failing? @thongsav-usgs